sstable/block.go

```diff
@@ -633,6 +641,9 @@ func (i *blockIter) readFirstKey() error {
 		i.firstUserKey = nil
 		return base.CorruptionErrorf("pebble/table: invalid firstKey in block")
 	}
+	if i.SyntheticPrefix != nil {
```

Comment:
nit: in other call sites, you have added the prefix before the `DecodeInternalKey` call. I don't think it matters, but for consistency, perhaps move this up.